src/reachability/box.jl

```diff
@@ -0,0 +1,39 @@
+"""
+    Box <: AbstractSolver
+
+Box performs over-approximated reachability analysis to compute the over-approximated output reachable set for a network.
+
+# Problem requirement
+1. Network: any depth, ReLU activation (more activations to be supported in the future)
+2. Input: Hyperrectangle
+3. Output: Hyperrectangle
+
+# Return
+`ReachabilityResult`
+
+# Method
+Reachability analysis using using boxes.
+
+# Property
+Sound but not complete?.
+
+
+"""
+struct Box <: AbstractSolver end
+
+function solve(solver::Box, problem::Problem)
+    reach = forward_network(solver, problem.network, problem.input)
+    return check_inclusion(reach, problem.output)
+end
+
+forward_layer(solver::Box, layer::Layer, inputs::Vector{<:LazySet}) = forward_layer.(solver, layer, inputs)
+
+function forward_layer(solver::Box, layer::Layer, input::AbstractPolytope)
+    return forward_layer(solver, layer, overapproximate(input, Hyperrectangle))
+end
+
+function forward_layer(solver::Box, layer::Layer, input::Hyperrectangle)
+    outlinear = overapproximate(affine_map(layer, input), Hyperrectangle)
```

Comment:
since `Box` is anyways taking the `Hyperrectangle` overapproximation, shouldn't this be

```suggestion
    outlinear = overapproximate(AffineMap(layer, input), Hyperrectangle)
```
where `AffineMap` means taking the lazy affine map of the layer, ie. `W * input + b`.

`affine_map` also works but creates a zonotope (as it takes the concrete affine map), then overapproximates that zonotope, while using the lazy affine map creates the hyperrectangle directly.

Reply:
in fact we can use https://github.com/JuliaReach/LazySets.jl/pull/2049 which is even faster.